### ISSUE TITLE
The aarch64 zig binaries segfault when building the posix-zig npm package.

### ISSUE THREAD

**Issue by williamstein** [ziglang/zig#12582]
### Zig Version

0.10.0-dev.3672+cd5a9ba1f

### Steps to Reproduce

I'm using an M1 mac and on everything nontrivial that I try to build, the zig aarch64 binaries segfault.  In particular, my npm package posix-zig is a good example.

Here's a quick summary:

1. Download the aarch64 macos or linux binary, e.g., zig-linux-aarch64-0.10.0-dev.3672+cd5a9ba1f.tar.xz, zig-macos-x86_64-0.10.0-dev.3672+cd5a9ba1f.tar.xz
2. Try to build posix-zig (details below).
3. Quickly get segfaults (on both linux and mac).
4. Try with an x86_64 binary, and it works fine.

More details:
```sh
~/tmp/python-wasm/packages/posix-zig$ uname -a
Darwin max.local 21.6.0 Darwin Kernel Version 21.6.0: Wed Aug 10 14:28:23 PDT 2022; root:xnu-8020.141.5~2/RELEASE_ARM64_T6000 arm64
~/tmp$ git clone https://github.com/sagemathinc/python-wasm
~/tmp$ cd python-wasm/packages/posix-zig
~/tmp/python-wasm/packages/posix-zig$ zig version
0.10.0-dev.3629+2ccaa5414
~/tmp/python-wasm/packages/posix-zig$ make
...
This works fine and builds some interesting zig code that provides missing posix functionality to nodejs...
...
```

Now try with zig downloaded just now from https://ziglang.org/download/:

```sh
~/tmp/python-wasm/packages/posix-zig$ make clean
~/tmp/python-wasm/packages/posix-zig$ wget https://ziglang.org/builds/zig-macos-x86_64-0.10.0-dev.3672+cd5a9ba1f.tar.xz; tar xf zig-macos-x86_64-0.10.0-dev.3672+cd5a9ba1f.tar.xz
~/tmp/python-wasm/packages/posix-zig$ export PATH=`pwd`/zig-macos-x86_64-0.10.0-dev.3672+cd5a9ba1f/:$PATH
~/tmp/python-wasm/packages/posix-zig$ make 
THIS works -- it's using x86_64 emulation.
~/tmp/python-wasm/packages/posix-zig$ wget https://ziglang.org/builds/zig-macos-aarch64-0.10.0-dev.3672+cd5a9ba1f.tar.xz; tar xf zig-macos-aarch64-0.10.0-dev.3672+cd5a9ba1f.tar.xz
~/tmp/python-wasm/packages/posix-zig$ make clean; make
~/tmp/python-wasm/packages/posix-zig$ make  # NOW WE GET THE SEGFAULT
...
zig build-lib -fstage1 -mcpu=baseline -OReleaseSmall --cache-dir /Users/wstein/build/cocalc/src/data/projects/2c9318d1-4f8b-4910-8da7-68a965514c95/tmp/python-wasm/packages/posix-zig/build/zig-cache -dynamic -lc -isystem /Users/wstein/build/cocalc/src/data/projects/2c9318d1-4f8b-4910-8da7-68a965514c95/tmp/python-wasm/packages/posix-zig/build/headers/include/node /Users/wstein/build/cocalc/src/data/projects/2c9318d1-4f8b-4910-8da7-68a965514c95/tmp/python-wasm/packages/posix-zig/src/lib.zig -fallow-shlib-undefined -target aarch64-macos -femit-bin=dist/aarch64-macos.node
make: *** [dist/aarch64-macos.node] Segmentation fault: 11
```

I also tested on aarch64 LInux and it similarly segfaults.

I built zig from source using https://github.com/ziglang/zig-bootstrap with the command `time ./build -j4 aarch64-macos-none native ` and there are two binaries I tried:

The one in `zig-bootstrap/out/host/bin` actually works!  

The one in `zig-bootstrap/out/zig-aarch64-macos-none-native/bin` segfaults.

I don't know what the difference is between these binaries, except they are both aarch64 binaries.

### Expected Behavior

Zig doesn't segfault doing basic things on aarch64.

### Actual Behavior

See above.  Zig segfaults on aarch64, except in one case when I built from source.

**Comment by williamstein** [comment]
This is now fixed in 0.10.0-dev.3880+e2bb92b2e

I think this was the same problem as https://github.com/ziglang/zig/issues/12483, since I was using @cImport on aarch64.

**Comment by williamstein** [comment]
Not only does the latest stage1 work fine to build my code, I was quickly able to port it to build with stage2 instead.   The build time then went from 6.727s (with stage1) to  0m3.791s (with stage2).  Yeah!
